packages/contracts/contracts/libs/v0.8.x/minter-libs/ERC20Lib.sol

```diff
@@ -0,0 +1,187 @@
+            "ETH is only null address"
+        );
+        _projectCurrencyConfig.currencySymbol = _currencySymbol;
+        _projectCurrencyConfig.currencyAddress = _currencyAddress;
+    }
+
+    /**
+     * @notice Processes payment for a fixed price project, using either ETH or
+     * ERC20 tokens as payment. If ERC20 tokens are used, the contract must be
+     * approved to spend at least `_pricePerTokenInWei` of `_currencyAddress`,
+     * which is validated in this function.
+     * @dev This function relies on msg.sender and msg.value, so it must be
+     * called directly from the contract that is receiving the payment.
+     * @param _projectCurrencyConfig ProjectCurrencyConfig of the project.
+     * @param _pricePerTokenInWei Price per token in wei.
+     * @param _projectId Project ID.
+     * @param _coreContract Core contract address.
+     * @param _isEngine True if core contract is an engine contract.
+     */
+    function processFixedPricePayment(
+        ProjectCurrencyConfig storage _projectCurrencyConfig,
+        uint256 _pricePerTokenInWei,
+        uint256 _projectId,
+        address _coreContract,
+        bool _isEngine
+    ) internal {
+        address currencyAddress = _projectCurrencyConfig.currencyAddress;
+        if (currencyAddress != address(0)) {
+            // ERC20 token is used for payment
+            require(msg.value == 0, "ERC20: No ETH when using ERC20");
+            validateERC20Approvals({
+                _msgSender: msg.sender,
+                _currencyAddress: currencyAddress,
+                _pricePerTokenInWei: _pricePerTokenInWei
+            });
+            SplitFundsLib.splitFundsERC20({
+                projectId: _projectId,
+                pricePerTokenInWei: _pricePerTokenInWei,
+                currencyAddress: currencyAddress,
+                coreContract: _coreContract,
+                _isEngine: _isEngine
+            });
+        } else {
+            // ETH is used for payment
+            require(
+                msg.value >= _pricePerTokenInWei,
+                "ETH: Min value to mint req."
+            );
+            SplitFundsLib.splitFundsETH({
+                projectId: _projectId,
+                pricePerTokenInWei: _pricePerTokenInWei,
+                coreContract: _coreContract,
+                _isEngine: _isEngine
+            });
+        }
+    }
+
+    /**
+     * Validate that `_msgSender` has approved the contract to spend at least
+     * `_pricePerTokenInWei` of `_currencyAddress` ERC20 tokens, and that
+     * `_msgSender` has a balance of at least `_pricePerTokenInWei` of
+     * `_currencyAddress` ERC20 tokens.
+     * Reverts if insufficient allowance or balance.
+     * @param _msgSender Address of the message sender to validate.
+     * @param _currencyAddress Address of the ERC20 token to validate.
+     * @param _pricePerTokenInWei Price per token in wei to validate.
+     */
+    function validateERC20Approvals(
```

Comment:
Nit: am I trippin that with our updated "no modifiers" convention, that this should be prefixed with an `_`?

Reply:
we usually prefix `internal` functions in our contracts with `_`, including the "no modifiers" convention. However, for internal functions of a library, we haven't been using underscores. These patterns are essentially the same as OpenZeppelin conventions.

Reply:
I see I see – SGTM ser.